File: dotnet/Logify.Alert.Core/Collectors/CustomDataCollector.cs
using System;
using System.Collections.Generic;

namespace DevExpress.Logify.Core {
    public class CustomDataCollector : IInfoCollector {
        readonly IDictionary<string, string> customData;

        public CustomDataCollector(IDictionary<string, string> customData, IDictionary<string, string> additionalCustomData) {
            this.customData = MergeData(customData, additionalCustomData);
        }

        IDictionary<string, string> MergeData(IDictionary<string, string> customData, IDictionary<string, string> additionalCustomData) {
            try {
                if (customData == null || customData.Count <= 0)
                    return additionalCustomData;
                if (additionalCustomData == null || additionalCustomData.Count <= 0)
                    return customData;

                Dictionary<string, string> result = new Dictionary<string, string>(customData);
                foreach (KeyValuePair<string, string> item in additionalCustomData)
                    result[item.Key] = item.Value;
                return result;
            }
            catch {
                return customData;
            }
        }

        public void Process(Exception ex, ILogger logger) {
            if (customData == null || customData.Count <= 0)
                return;

            logger.BeginWriteObject("customData");
            foreach (string key in customData.Keys)
                logger.WriteValue(key, customData[key]);
            logger.EndWriteObject("customData");
        }
    }
}

File: java/Core/src/main/java/com/devexpress/logify/alert/core/collectors/CustomDataCollector.java
package com.devexpress.logify.alert.core.collectors;

import com.devexpress.logify.alert.core.logger.ILogger;
import java.util.Map;

public class CustomDataCollector implements IInfoCollector {

    private Map<String, String> customData;

    public  CustomDataCollector(Map<String, String> customData){
            this.customData = customData;
    }

    public void process(Throwable ex, ILogger logger) {
        if (customData == null || customData.size() == 0)
            return;

        logger.beginWriteObject("customData");
        for (String key : customData.keySet())
            logger.writeValue(key, customData.get(key));
        logger.endWriteObject("customData");
    }
}